### DISCUSSION TITLE
[SOLVED] New features : RTL feed direction and Calendar

### DISCUSSION THREAD

**Discussion by RachidTagzen** [martinrotter/rssguard#1018] [Ideas]
I tried to contribute to RSS Guard by adding 2 features, that I already contributed to [QuiteRSS](https://github.com/QuiteRSS/quiterss/pull/1115) years ago. But I got this error when I clicked "Create pull request" : 
![error](https://github.com/martinrotter/rssguard/assets/25085777/9c78317e-5d0b-4507-95a2-639b17185a80)

So, this is the description of my contribution :

I added 2 features:
    - RTL layout direction to feeds
    - A calendar to avoid old articles

---
**RTL**
This feature lets change the feed's articles from right to left for languages such as Arabic, Urdu, Hebrew, Farsi...
To enable it :  The desired feed -> Edit selected item -> Articles -> Right-to-left layout.

![LTR](https://github.com/martinrotter/rssguard/assets/25085777/a555d38d-a066-44d2-b0d0-158daae4ac2b)
![RTL](https://github.com/martinrotter/rssguard/assets/25085777/81cde3e3-f645-4c2b-a46a-d60aa5f54bd7)

---
**Calendar**
This feature lets avoid downloading old articles to preserve a small database. So, if we cleanup the database, we can ensure that the articles with a date old than the chosen date (as above) will not be added to the database.
This works also with a new empty database.

To enable it :
  - Set a general date for all feeds (in Tools -> Settings -> Feeds & articles).
  - Set a date for a desired single feed (in The desired feed -> Edit selected item -> Articles).

![Datetime - Feed](https://github.com/martinrotter/rssguard/assets/25085777/28876841-95b9-43e4-a5df-2f2e7c226a7a)
![Datetime - Global](https://github.com/martinrotter/rssguard/assets/25085777/5279a0f4-6370-4e93-8f50-40271e5b276e)

**Comment by martinrotter** [comment]
Hello. Thanks for contribution but it seems there must be something wrong on your side as many non-collaborators have created pull requests just fine, see here.

https://github.com/martinrotter/rssguard/pulls?q=is%3Apr+is%3Aclosed

Try again, maybe you do something wrong?

https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request

Also, please make sure to ideally separate your two contributions into two PRs, if that is not possible due to "scrumbled" commits, do not worry its okay.

**Comment by RachidTagzen** [comment]
Thank you.
I separate the 2 features, then I tried to create a a pull request.
Sadly, the same issue: "must be a collaborator".

I seeked for a solution, and I found that probably you should verify and change this in your account :

---
![Firefox_Screenshot_2023-07-22T14-06-29 825Z](https://github.com/martinrotter/rssguard/assets/25085777/f411ed2d-a0b8-4b3e-a168-434220465096)

---
https://chat.openai.com/share/e6f25816-9949-460b-ac7e-50945be64d64

https://docs.github.com/en/organizations/managing-user-access-to-your-organizations-repositories/adding-outside-collaborators-to-repositories-in-your-organization

https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/managing-repository-settings/managing-teams-and-people-with-access-to-your-repository#inviting-a-team-or-person

**Comment by RachidTagzen** [comment]
I forked the repository, then I created the pull requests.

[
New Features: **Avoid Downloading Old Articles to Preserve a Small Database**    and    **Supporting RTL**](https://github.com/martinrotter/rssguard/pull/1019/commits)
